stablehlo/tests/interpret_sort.mlir

```diff
@@ -0,0 +1,35 @@
+// RUN: stablehlo-translate --interpret -split-input-file %s
```

Comment:
Can we re-write the example in such a way that sort and stable-sort would differ? With our offline discussion it seem you have a plan. Let's try that.

Reply:
As I am thinking more about this:
If the solution to this involves sorting different representations of NaN (using partial order), then we might avoid (also ignore my comment above) that mainly because seems more of a testing-interesting-feature to me where we have to rely on comparison semantics of  NaN. Code-coverage wise we will not gain anything.

Reply:
There is no way to sort primitive types to tell the difference. Since we didn't implement `TOTALORDER` in CompareOp, we also cannot sort between different representations of NaNs. Sorting should (rightfully) be broken with NaNs as we are using unordered comparisons.